Engine/source/environment/waterObject.h

```diff
@@ -156,33 +156,33 @@ class WaterObject : public SceneObject
    static void initPersistFields();
 
    // SimObject
-   virtual bool onAdd();
-   virtual void onRemove();
-   virtual void inspectPostApply();
-   virtual bool processArguments(S32 argc, ConsoleValue *argv);
+   bool onAdd();
+   void onRemove();
+   void inspectPostApply();
+   bool processArguments(S32 argc, ConsoleValue *argv);
```

Comment:
This is going to break `WaterPlane` class. Please pay attention as to what you're doing here.

All SimObject methods like this should always be virtual when the object might be inherited in the future.
Same goes for linage of `NetObject`, as I'm seeing the same for `packUpdate`/`unpackUpdate` methods. and other commonly inherited classes.

Do not trust Visual Studio's static analyzers without manually reviewing each case because it is often very wrong.

Reply:
i have tested all the classes i have changed by loading up an instance of them in the world editor and they work fine is there another test i should do?